modules/light-clients/08-wasm/types/wasm.go

```diff
@@ -6,29 +6,33 @@ import (
 	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/internal/ibcwasm"
 )
 
+// CodeHash is a type alias used for wasm byte code checksums.
+type CodeHash []byte
+
 // GetAllCodeHashes is a helper to get all code hashes from the store.
 // It returns an empty slice if no code hashes are found
-func GetAllCodeHashes(ctx sdk.Context) ([][]byte, error) {
+func GetAllCodeHashes(ctx sdk.Context) ([]CodeHash, error) {
 	iterator, err := ibcwasm.CodeHashes.Iterate(ctx, nil)
 	if err != nil {
 		return nil, err
 	}
 
-	codeHashes, err := iterator.Keys()
+	keys, err := iterator.Keys()
 	if err != nil {
 		return nil, err
 	}
 
-	if codeHashes == nil {
-		codeHashes = [][]byte{}
+	var codeHashes []CodeHash
+	for _, key := range keys {
+		codeHashes = append(codeHashes, key)
 	}
```

Comment:
I removed this check as I figured `nil` should be a valid return value. From what I gathered the only place needing this was in `genesis_test.go` [here](https://github.com/cosmos/ibc-go/pull/5004/files#diff-5a718389573601b8fa391f2362feabf88a431f525e6adc2971c6f1f3f60b2619L62).

Reply:
It's an either or I guess but in general not having a possibly `nil` value floating around is a slight preference. not a hill I'm dying on though